### DISCUSSION TITLE
Installation script: Trouble installing paperless-ng

### DISCUSSION THREAD

**Discussion by BlueBarracuda22** [jonaswinkler/paperless-ng#1129] [Support]
Paperless newbie here. I'm trying to setup paperless-ng for the first time using the installation script. I have the latest version of Docker installed. I have tried to follow the directions on [readthedocs](https://paperless-ng.readthedocs.io/en/latest/setup.html#install-paperless-from-docker-hub-using-the-installation-script), but am getting stuck at the first step. Please see my below terminal output. Thanks

`Last login: Sat Jun 19 16:43:26 on ttys000
username@MacBook-Pro ~ % curl -L https://raw.githubusercontent.com/jonaswinkler/paperless-ng/master/install-paperless-ng.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8441  100  8441    0     0  37852      0 --:--:-- --:--:-- --:--:-- 37683
bash: line 76: timedatectl: command not found

############################################
############################################

This script will download, configure and start paperless-ng.

1. Folder configuration
=======================

The target folder is used to store the configuration files of 
paperless. You can move this folder around after installing paperless.
You will need this folder whenever you want to start, stop, update or 
maintain your paperless instance.

The consume folder is where paperles will search for new documents.
Point this to a folder where your scanner is able to put your scanned
documents.

CAUTION: You must specify an absolute path starting with / or a relative 
path starting with ./ here. Examples:
  /mnt/consume
  ./consume

Invalid folder: CONSUME_FOLDER=$ask_result

The media folder is where paperless stores your documents.
Leave empty and docker will manage this folder for you.
Docker usually stores managed folders in /var/lib/docker/volumes.

CAUTION: If specified, you must specify an absolute path starting with /
or a relative path starting with ./ here.

Invalid folder: MEDIA_FOLDER=$ask_result

The data folder is where paperless stores other data, such as your
SQLite database (if used), the search index and other data.
As with the media folder, leave empty to have this managed by docker.

CAUTION: If specified, you must specify an absolute path starting with /
or a relative path starting with ./ here.

Invalid folder: DATA_FOLDER=$ask_result

2. Application configuration
============================

The port on which the paperless webserver will listen for incoming
connections.

Paperless requires you to configure the current time zone correctly.
Otherwise, the dates of your documents may appear off by one day,
depending on where you are on earth.

Database backend: PostgreSQL and SQLite are available. Use PostgreSQL
if unsure. If you're running on a low-power device such as Raspberry
Pi, use SQLite to save resources.

Invalid option: DATABASE_BACKEND=$ask_result

Paperless is able to use Apache Tika to support Office documents such as
Word, Excel, Powerpoint, and Libreoffice equivalents. This feature
requires more resources due to the required services.

Invalid option: TIKA_ENABLED=$ask_result

Specify the default language that most of your documents are written in.
Use ISO 639-2, (T) variant language codes: 
https://www.loc.gov/standards/iso639-2/php/code_list.php
Common values: eng (English) deu (German) nld (Dutch) fra (French)

Specify the user id and group id you wish to run paperless as.
Paperless will also change ownership on the data, media and consume
folder to the specified values, so it's a good idea to supply the user id
and group id of your unix user account.
If unsure, leave default.

3. Login credentials
====================

Specify initial login credentials. You can change these later.
A mail address is required, however it is not used in paperless. You don't
need to provide an actual mail address.

Password cannot be empty.

Passwords did not match

Password cannot be empty.

Passwords did not match

Passwords did not match

Password cannot be empty.

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Password cannot be empty.

Passwords did not match

Password cannot be empty.

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Password cannot be empty.

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Password cannot be empty.

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

Passwords did not match

username@MacBook-Pro ~ % 
`

**Comment by jonaswinkler** [comment]
I should add a note that this script is for Linux only.

**Reply by jumpingbrowntrout** [reply]
@jonaswinkler I am having the same issue Ubuntu 18.04 (Albeit WSL)

It appears as though the "ask" command isn't  prompting and the script keeps running as though 'enter' is continually pressed.

Interestingly if I wget the install script (install-paperless-ng.sh) and run it locally I get the "ask" prompts but throws a "not found" errors.

```
me@DESKTOP-CM4ID10:~$ sh install-paperless-ng.sh
install-paperless-ng.sh: 45: install-paperless-ng.sh.1: [[: not found
install-paperless-ng.sh: 50: install-paperless-ng.sh.1: [[: not found
install-paperless-ng.sh: 55: install-paperless-ng.sh.1: [[: not found
install-paperless-ng.sh: 60: install-paperless-ng.sh.1: [[: not found
timedatectl: invalid option -- 'p'

############################################
###   Paperless-ng docker installation   ###
############################################

This script will download, configure and start paperless-ng.

1. Folder configuration
=======================

The target folder is used to store the configuration files of
paperless. You can move this folder around after installing paperless.
You will need this folder whenever you want to start, stop, update or
maintain your paperless instance.

install-paperless-ng.sh. 5: install-paperless-ng.sh.1: [[: not found
Target folder () [/home/mrlb/paperless-ng]:
install-paperless-ng.sh: 10: install-paperless-ng.sh.1: [[: not found
install-paperless-ng.sh: 15: install-paperless-ng.sh.1: [[: not found
install-paperless-ng.sh: 15: install-paperless-ng.sh.1: Bad substitution

```

**Reply by iwconfig** [reply]
@browntownington have a look at #1136. And to answer why the second attempt didn't work: you used `sh` instead of `bash`.

**Reply by iwconfig** [reply]
Also, consider having a look at #1137 because I see you too have a version of `systemd` that is too old.

**Reply by jumpingbrowntrout** [reply]
@iwconfig thank you!!! note to self don't try troubleshooting at 1am. 
Both of these suggestions resolved my issues and installation was then BREEZE :)

**Reply by iwconfig** [reply]
Happy to help!

**Reply by iwconfig** [reply]
@jonaswinkler I don't see any reason why this shouldn't work under OSX/MacOS with some small adjustments. It seems you only have to

- use `curl` instead of `wget`,
- implement the changes I propose in #1136 and #1137,
- and fix `tr: Illegal byte sequence` error by changing

  https://github.com/jonaswinkler/paperless-ng/blob/4718fe13c2fb0139d7a7dee1ff53b3514633a952/install-paperless-ng.sh#L283

  to

      SECRET_KEY=$(cat /dev/urandom | LC_ALL=C tr -dc 'a-zA-Z0-9' | fold -w 64 | head -n 1)

I don't have a Mac so I don't know for sure so if anyone with one would like to confirm this I can put up a PR.

---

EDIT: Never mind! Just now I noticed #1118 :+1:
